degirum_tools/classification_eval.py

```diff
@@ -0,0 +1,145 @@
+    accuracy = correct_predictions / total_data
+    return accuracy
+
+
+class ImageClassificationModelEvaluator:
+    def __init__(
+        self,
+        dg_model,
+        foldermap,
+        top_k,
+        output_confidence_threshold=0.001,
+        input_resize_method="bicubic",
+        input_pad_method="crop-last",
+        image_backend="opencv",
+        input_img_fmt="JPEG",
+    ):
+        """
+        Constructor.
+            This class computes the Top-k Accuracy for Classification models.
+
+            Args:
+                dg_model (Detection model): Classification model from the Degirum model zoo.
+                k (int) : The value of `k` in top-k.
+                foldermap (dict): The key represents integer (starting from 0) and values represent the class names (folder names) of the validation dataset.
+                                 - For example : Gender Classification model - foldermap = {0: "0", 1: "1"}
+                input_resize_method (str): Input Resize Method.
+                input_pad_method (str): Input Pad Method.
+                image_backend (str): Image Backend.
+                input_img_fmt (str): InputImgFmt.
+        """
+
+        self.dg_model = dg_model
+        self.foldermap = foldermap
+        self.top_k = top_k
+        if self.dg_model.output_postprocess_type == "Classification":
+            self.dg_model.output_confidence_threshold = output_confidence_threshold
+            self.dg_model.input_resize_method = input_resize_method
+            self.dg_model.input_pad_method = input_pad_method
+            self.dg_model.image_backend = image_backend
+            self.dg_model.input_image_format = input_img_fmt
+        else:
+            raise Exception("Model loaded for evaluation is not a Classification Model")
+
+    @classmethod
+    def init_from_yaml(cls, dg_model, config_yaml):
+        with open(config_yaml) as f:
+            args = yaml.load(f, Loader=yaml.FullLoader)
+
+        return cls(
+            dg_model=dg_model,
+            foldermap=args["foldermap"],
+            top_k=args["top_k"],
+            output_confidence_threshold=args["output_confidence_threshold"],
+            input_resize_method=args["input_resize_method"],
+            input_pad_method=args["input_pad_method"],
+            image_backend=args["image_backend"],
+            input_img_fmt=args["input_img_fmt"],
+        )
+
+    def evaluate(
+        self,
+        image_folder_path: str,
+        print_frequency: int = 0,  # print_frequency (int): Number of image batches to be evaluated at a time.
+    ):
+        total_data = 0
+        total_misclassified = 0
+        misclassified_count_dct = {}
+
+        for label in os.listdir(image_folder_path):
+            gndtruth = []
+            data_per_class = count_files_in_directory(
+                os.path.join(image_folder_path, label)
+            )
+            total_data += data_per_class
```

Comment:
total data should be counted when processing the images one by one
That is why count_files_in_directory is not needed and error prune